pydm/widgets/drawing.py

```diff
@@ -1184,3 +1185,34 @@ def resetPoints(self):
         self.update()
 
     points = Property("QStringList", getPoints, setPoints, resetPoints)
+
+
+class PyDMDrawingIrregularPolygon(PyDMDrawingPolyline):
+    """
+    A widget contains an irregular polygon (arbitrary number of vertices, arbitrary lengths).
+
+    This is a special case of the PyDMDrawingIrregularPolygon, adding the requirement that 
+    the last point is always identical to the first point.
+
+    This widget is created for compatibility with MEDM's *polygon* widget.
+
+    Parameters
+    ----------
+    parent : QWidget
+        The parent widget for the Label
+    init_channel : str, optional
+        The channel to be used by the widget.
+    """
+
+    def setPoints(self, points):
```

Comment:
I think the `points` property in designer is routing around this entirely? Perhaps the definition from polyline is used instead due to the way the property is defined above? We might need to redefine the `points` property

Reply:
It's a new problem with the internal representation of `self._points`.  With this PR, that representation was changed from a string to a python list or tuple and that is producing a failure.  Inserting some diagnostic print statements has exposed the problem when the second (perfectly valid) point is added:

```python
DIAGNOSTIC (pydm.widgets.drawing,setPoints): verified=[[1.0, 1.0], [1.0, 2.0]]
DIAGNOSTIC (pydm.widgets.drawing,getPoints): self._points=[[1.0, 1.0], [1.0, 2.0]]
Exception occurred while running Qt Designer.
TypeError: unable to convert a Python 'list' object to a C++ 'QStringList' instance
```

Reply:
In next version for review.